addons/survey/models/survey.py

```diff
@@ -0,0 +1,1096 @@
+            answer_candidates = dict_keys_startswith(post, answer_tag)
+            comment_flag = answer_candidates.pop(("%s_%s" % (answer_tag, -1)), None)
+            if question.comments_allowed:
+                comment_answer = answer_candidates.pop(("%s_%s" % (answer_tag, 'comment')), '').strip()
+            # Preventing answers with blank value
+            if all([True if answer.strip() == '' else False for answer in answer_candidates.values()]):
+                errors.update({answer_tag: question.constr_error_msg})
+            # There is no answer neither comments (if comments count as answer)
+            if not answer_candidates and question.comment_count_as_answer and (not comment_flag or not comment_answer):
+                errors.update({answer_tag: question.constr_error_msg})
+            # There is no answer at all
+            if not answer_candidates and not question.comment_count_as_answer:
+                errors.update({answer_tag: question.constr_error_msg})
+        return errors
+
+    def validate_matrix(self, question, post, answer_tag):
+        errors = {}
+        if question.constr_mandatory:
+            lines_number = len(question.labels_ids_2)
+            answer_candidates = dict_keys_startswith(post, answer_tag)
+            # Number of lines that have been answered
+            if question.matrix_subtype == 'simple':
+                answer_number = len(answer_candidates)
+            elif question.matrix_subtype == 'multiple':
+                answer_number = len(set([sk.rsplit('_', 1)[0] for sk in answer_candidates.keys()]))
+            else:
+                raise RuntimeError("Invalid matrix subtype")
+            # Validate that each line has been answered
+            if answer_number != lines_number:
+                errors.update({answer_tag: question.constr_error_msg})
+        return errors
+
+
+class SurveyLabel(models.Model):
+    ''' A suggested answer for a question '''
+    _name = 'survey.label'
+    _rec_name = 'value'
+    _order = 'sequence,id'
+    _description = 'Survey Label'
+
+    question_id = fields.Many2one('survey.question', string='Question',
+        ondelete='cascade')
+    question_id_2 = fields.Many2one('survey.question', string='Question',
+        ondelete='cascade')
+    sequence = fields.Integer(string='Label Sequence order', default=10)
+    value = fields.Char(string="Suggested value", translate=True,
+        required=True)
+    quizz_mark = fields.Float(string='Score for this choice', help="A positive score indicates a correct choice; a negative or null score indicates a wrong answer")
+
+    @api.multi
+    @api.constrains('question_id', 'question_id_2')
+    def _check_question_not_empty(self):
+        '''Ensure that field question_id XOR field question_id_2 is not null'''
+        for label in self:
+            # 'bool()' is required in order to make '!=' act as XOR with objects
+            if not bool(label.question_id) != bool(label.question_id_2):
```

Comment:
not (bool(label.question_id) != bool(label.question_id_2)) looks more readable to me